crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -76,10 +76,37 @@ impl Operations {
         self.operations.retain(|x| x.name.ne(&op_name));
     }
 
-    pub fn try_new(dir: impl AsRef<Path>, cloud_name: &str) -> Result<Self, OperationsError> {
+    pub fn try_new(
+        dir: impl AsRef<Path>,
+        cloud_name: Option<&str>,
+    ) -> Result<Self, OperationsError> {
         get_operations(dir.as_ref(), cloud_name)
     }
 
+    pub fn get_child_ops(
+        ops_dir: impl AsRef<Path>,
+    ) -> Result<HashMap<String, Self>, OperationsError> {
+        let mut child_ops: HashMap<String, Operations> = HashMap::new();
+        let child_entries = fs::read_dir(&ops_dir)
+            .map_err(|_| OperationsError::ReadDirError {
+                dir: ops_dir.as_ref().into(),
+            })?
+            .map(|entry| entry.map(|e| e.path()))
+            .collect::<Result<Vec<PathBuf>, _>>()?
+            .into_iter()
+            .filter(|path| path.is_dir())
+            .collect::<Vec<PathBuf>>();
+        for cdir in child_entries {
+            let ops = Operations::try_new(cdir.clone(), None)?;
+            if let Some(id) = cdir.iter().next_back() {
+                if id.to_str().is_some() {
+                    child_ops.insert(id.to_str().unwrap().into(), ops);
```

Comment:
To avoid that `unwrap`, I'd replace this `if` block with an `if let` block for the `Some(op_name)` variant.

Reply:
done